### PR TITLE
CPLAT-9092 Add built_redux to Redux Docs

### PR DESCRIPTION
**Pull request by joebingham-wk**
## Motivation
  
To aid in the transition to Redux, we want to add a doc that outlines the differences between built_redux and Redux, along with tips on how to transition.
## Changes
  
- Add a built_redux_to_redux markdown doc
#### Release Notes
  
- Add a built_redux to OverReact Redux transition guide.
## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 
@aaronlademann-wf @greglittlefield-wf @willdrach-wk @sydneyjodon-wk
### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: doc/built_redux_to_redux.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+        import './components/component.dart';
+
+        main() {
+            setClientConfiguration();
+            react_dom.render(
+                (ReduxProvider()..store = randomColorStore)(
+                    ConnectedComponent()(),
+                ),
+                querySelector('#content'));
+        }
```
Comment (aaronlademann-wf) [inline doc/built_redux_to_redux.md]: This indentation doesn't match the indentation of the ``` backticks

File: doc/built_redux_to_redux.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+
+        ```dart
+        // Simple built_redux component
+        // Assume there is a store with a state field `text` and an action `updateText`.
+        @Factory()
+        UiFactory<SimpleProps> Simple = _$Simple;
+
+        @Props()
+        class _$SimpleProps extends UiProps {
+            Store<App, AppBuilder, AppActions> store;
```
Comment (aaronlademann-wf) [inline doc/built_redux_to_redux.md]:
```suggestion
          Store<App, AppBuilder, AppActions> store;
```

File: doc/built_redux_to_redux.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+            render() {
+                return (
+                    Fragment()(
+                        // Assume there is a function `randomString` that generates random text
+                        (Dom.button()..onClick => props.updateText(randomString()))('Change Text'),
+                        (Dom.div())(props.text)
+                    );   
+                );
+            }
+        }
```
Comment (aaronlademann-wf) [inline doc/built_redux_to_redux.md]: All of these 4-space indentations can/should be converted to 2-space.

File: doc/built_redux_to_redux.md [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+    ReduxState.from(customActionValue: action.customActionValue, secondCustomActionValue: action.seconCusomActionValue);
+final simpleActionReducer = (ReduxState prevState, SimpleAction action) => 
+    ReduxState.from(simpleActionState: !prevState.simpleActionState);
+
+Reducer<ReduxState> countReducer = combineReducers([
+    TypedReducer<ReduxState, IncremementAction>(incrementReducer),
+    TypedReducer<ReduxState, DecrementAction>(decrementReducer),
+    TypedReducer<ReduxState, CustomAction>(customActionReducer),
+    TypedReducer<ReduxState, SimpleAction>(simpleActionReducer),
+]);
```
[the remark this answers is not in the document]
Comment (joebingham-wk) [reply doc/built_redux_to_redux.md]: Yep, good catch. The reason they were all there was that `combineReducers` used to be part of the the Reducer section below, but then I moved it into its own section and just grabbed the `ReducerBuilder` example. So no real reason.

**Pull request merged**